site/app/models/gradeable/Gradeable.php

```diff
@@ -491,12 +491,42 @@ public function setPeerGradersList($input) {
         else {
             $this->core->getQueries()->clearPeerGradingAssignment($this->getId());
             foreach ($input as $row_num => $vals) {
-                $this->core->getQueries()->insertPeerGradingAssignment($vals["grader"], $vals["student"], $this->getId());
+                $this->core->getQueries()->insertPeerGradingAssignment($vals["grader"], $vals["student"], $vals["feedback"]);
                 $this->modified = true;
                 $this->peer_grading_pairs = $this->core->getQueries()->getPeerGradingAssignment($this->getId());
             }
         }
     }
+    
+    public function setPeerFeedback($grader_id, $student_id, $feedback_full, $feedback_id) {
+        $bad_input = [];
+        if ($this->core->getQueries()->getUserById($grader_id) == null) {
+            array_push($bad_input, ($grader_id));
+        }
+        if ($this->core->getQueries()->getUserById($student_id) == null) {
+            array_push($bad_input, ($student_id));
+        }
+        if (!empty($bad_input)) {
+            $msg = "The given user id is not valid: ";
+            array_walk($bad_input, function ($val) use (&$msg) {
+                $msg .= " {$val}";
+            });
+            $this->core->addErrorMessage($msg);
+        }
+        else {
+            $this->core->getQueries()->insertPeerGradingFeedback($grader_id, $student_id, $this->getId(), $feedback_full, $feedback_id);
+        }
+        var_dump($this->core->getQueries()->getPeerFeedback($this->getId()));
+    }
+    
+    public function getPeerFeedback($grader_id, $anon_id) {
+        $feedback = $this->core->getQueries()->getPeerFeedback($this->getId());
```

Comment:
Perhaps add a database function `getPeerFeedback($user_id, $is_team)` which returns the results for a single team/user_id.

Reply:
Added this function: getPeerFeedbackInstance